src/components/KFileUpload/KFileUpload.vue

```diff
@@ -0,0 +1,344 @@
+<template>
+  <div
+    class="k-file-upload w-100"
+  >
+    <KLabel
+      v-if="label && appearance === 'file'"
+      :for="customInputId"
+      v-bind="labelAttributes"
+      data-testid="k-file-upload-label"
+    >
+      {{ label }}
+    </KLabel>
+
+    <KInput
+      :id="customInputId"
+      ref="fileInput"
+      :key="fileInputKey"
+      type="file"
+      :accept="accept"
+      :help="help"
+      :max-file-size="maximumFileSize"
+      :class="[
+        'w-100',
+        'upload-input',
+        'cursor-pointer',
+        { 'image-upload': appearance === 'image' },
+        { 'input-error' : hasUploadError }
+      ]"
+      @change="onFileChange"
+    />
+
+    <slot name="icon">
+      <KIcon
+        v-if="appearance === 'image'"
+        :size="iconSize"
+        :color="iconColor"
+        :icon="icon"
+        class="image-upload-icon"
+        @click="updateFile"
+      />
+    </slot>
+
+    <p
+      v-if="appearance === 'image'"
+      class="image-upload-description"
+      @click="updateFile"
+    >
+      {{ placeholder }}
+    </p>
+    <KButton
+      v-if="fileValue && appearance === 'file' && removable"
+      id="pseudoCancel"
+      type="reset"
+      appearance="primary"
+      size="small"
+      data-testid="k-pseudo-cancel-button"
+      @click="resetInput"
+      @keyup.enter="resetInput"
+    >
+      <template #icon>
+        <KIcon
+          icon="close"
+          size="16"
+        />
+      </template>
+    </KButton>
+    <KButton
+      v-if="appearance === 'file'"
+      :appearance="buttonAppearance"
+      class="k-file-upload-btn"
+      data-testid="k-file-upload-button"
+      size="small"
+      @click="updateFile"
+      @keyup.enter="updateFile"
+    >
+      {{ buttonText }}
+    </KButton>
+    <span
```

Comment:
A `span` tag cannot gain focus by default. Can we utilize an anchor tag `<a>` instead?

Reply:
Why would you want to focus the text inside a input field, when that parent field can be focused ?